blog/content/post/2021-10-03-fobnail-promotion.md

```diff
@@ -0,0 +1,266 @@
+
+So in short, the main principles of Fobnail are:
+
+- to make the attestation as simple as possible (e.g. by using LED indicators
+  to inform a user about the decision of trustworthiness)
+- act as the axiomatically trusted device (an [iTurtle](https://www.usenix.org/legacy/event/hotsec07/tech/full_papers/mccune/mccune_html/index.html))
+  for the attestation process
+- provide attestation services without network connection
+
+So how does it improve situation for the mentioned projects? As long as the
+firmware is protected the risk is not so high that the components have been
+tampered with. But the attestation process over the network may be hijacked or
+return untrustworthy result. By using fobnail, whole attestation and decision
+process is moved to the tiny USB device so that the user can confirm the result
+visually. This solves the problem of untrusted networks or attestation servers.
+
+Secondly if we do not trust that the firmware is bug-free and provides reliable
+measurements one may use D-RTM when booting the target Linux kernel and perform
+the attestation based on D-TM measurements with Fobnail.
+
+Thirdly when we look at TOTP/HOTP the whole decision process of secret
+unsealing is done on the platform, also the platform is responsible to display
+or send the 6 digit code to the display or USB token. This creates a potential
+risk that the code has been faked and the measurements aren't really reliable.
+Fobnail removes such risk by moving the platform state evaluation on the token
+side. Before the Fobnail token becomes usable it must be provisioned first with:
+
+- Reference Integrity Measurement (RIM) Database (approved set of platform
+  measurements)
+- Attestation Policy Database (decision policies used to evaluate the
+  trustworthiness of the attestation data)
+- Attestation Protected Object (optional secret data that is going to be
+  unsealed on successful attestation process)
+- Identity and encryption certificates
+
+So Fobnail knows about all the known good platform states. The host
+responsibility in the attestation process is to provide the TPM quote (set of
+platform integrity measurements, event logs, etc.) signed by the attestation
+key to the token for evaluation. Then the decision is returned to the host and
+indicated to the platform owner with a physical or digital response. Fobnail is
+also a flexible device which allows to perform the attestation of multiple
+devices (by using multiple RIMs and policies).
+
+As you can see the decision process is now done in the secure environment
+unlike the example heads boot flow. This doesn't however resolve the problem of
+reliable measurements if the Static Root of Trust measurements are used. THere
+is still the risk that the measurement have been faked because firmware has
+been tampered with or was not protected. With the help comes the D-RTM. Imagine
```

Comment:
From my understanding of Microsoft public documentation, one of the usecases of D-RTM is forward sealing of S-RTM part: https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-defender-system-guard/how-hardware-based-root-of-trust-helps-protect-windows#secure-launchthe-dynamic-root-of-trust-for-measurement-drtm

Reply:
I'm not talking specifically about the management of SRTM measurements but rather about reliability of the measurements, i.e. if what was measured was the measurement of what was really executed. How do you see this information to be included here?

Reply:
@miczyg1 then I agree, but I would be more straight forward about expressing the difference between what was measured (load time measurement) vs runtime/execution time measurement. In that light I assume there is not competing technology to D-RTM, maybe SGX and similar but those require attestation with Intel server IIUC and also have other drawbacks.